Rebalanced Food/Removed Some Foods, Altered Boss difficulty, fixed end

Monydragon opened the pull request:
Renamed the project to Lost Dreams, Fixed Several things for balancing. Fixed end scene to go to menu after credits instead of returning back to the scene. Modified chests loot and drop tables for all enemies. Created more spiders, Added Big spider. cleaned some dialogue.

Monydragon committed:
Rebalanced Food/Removed Some Foods, Altered Boss difficulty, fixed end

Renamed the project to Lost Dreams, Fixed Several things for balancing. Fixed end scene to go to menu after credits instead of returning back to the scene. Modified chests loot and drop tables for all enemies. Created more spiders, Added Big spider. cleaned some dialogue.

zkrizo commented on Assets/_Game/Scripts/UI/UICredits.cs:
```diff
@@ -12,6 +12,6 @@ public void Setup(GameSystems systems)
     public void OnBackClicked()
     {
-        _systems.StateManager.NavigateToState(null, true);
+        _systems.StateManager.NavigateToState(typeof(MainMenuState), true);
```
This will lead to multiple mainmenustates in the queue. Better fix would be to do 2 transitions rather than 1 when you push the credits after the game. One to pop the current state (the game state) and transfer it back to main menu and then push the CreditState right after

PatrickMoffett committed:
Fixed Apartment Scene to Push MainMenu then credits from dialogue invoke

Also add bool to pop states when loading new level states.
Also Fixed End Game Song Theme
